src/librustc/mir/interpret/allocation.rs

```diff
@@ -15,16 +15,17 @@ use std::borrow::Cow;
 
 #[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Hash, RustcEncodable, RustcDecodable)]
 pub struct Allocation<Tag=(),Extra=()> {
-    /// The actual bytes of the allocation.
-    /// Note that the bytes of a pointer represent the offset of the pointer
-    pub bytes: Vec<u8>,
+    /// The byte allocation and definedness.
+    /// Bytes of fully undefined allocations are allocated lazily on potentially mutable changes.
+    pub alloc: Option<AllocationBytes>,
```

Comment:
`alloc` is usually used as name for something of type `Allocation`. Here and below, you are breaking with that pattern. Maybe call this `alloc_bytes`, or `alloc_content`? In the latter case the type should also change to `AllocationContent`.